src/components/CardAction/CardAction.js

```diff
@@ -1,54 +1,82 @@
-import React, {useState, useEffect} from 'react'
-import './CardAction.css'
-import checkIcon from '../../assets/icons/iconcheck.svg'
+import React, { useState, useEffect } from "react";
+import "./CardAction.css";
+import checkIcon from "../../assets/icons/iconcheck.svg";
 
+/**
+ *
+ * as the other components there is no need for the state and
+ * again should be better to use classes in css rather the styles
+ * props.
+ * In this case even better you can use the disable attribute of the button
+ * to define the style
+ *
+ * Here an example of how much easier this component could be
+ * ```
+ * export const CardAction = ({booked, bookable}) => (
+ *  <>
+ *      {booked > 0
+ *          ? (
+ *              <p className="cardActionBooked">
+ *                  <img src={checkIcon} alt="" />
+ *                  {`Booked for ${booked} day${booked > 1 ? 's' : ''}`}
+ *              </p>
+ *          ) : (
+ *              <button className="cardActionButton" disabled={!bookable}>
+ *                  {bookable ? 'Book' : 'Not bookable'}
+ *              </button>
+ *          )
+ *       }
+ *  </>
+ * )
+ */
```

Comment:
This reply applies to all the comments related with your state comments:

@huvber Makes total sense. Thank you for that specifically, this example you gave (in the CardAction component) is a brutal piece of learning.

That was the way my brain currently thinks react. However the example you shared made me see how different it can be.